### PR TITLE
Use Github action to upload to PyPI on new release

### PR DESCRIPTION
**Pull request by sfdye**
- Added `secrets.PYPI_PASSWORD` to project settings
- Use twine to build artifact and upload to PyPI on new release

TODO

- [ ] Test twine with new patch release (1.47.1)

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
           python-version: "${{ matrix.python-version }}"
       - name: "Install dependencies"
         run: |
-          pip install --upgrade pip
+          python -m pip install --upgrade pip
```
Comment (sfdye) [inline .github/workflows/check.yml:19]: this is preferred way by latest pip

File: .github/workflows/python-publish.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+name: Publish to PyPI
+on:
+  release:
+    types: [created]
```
Comment (sfdye) [inline .github/workflows/python-publish.yml:4]: Although I am not entirely sure the difference betwewen `created` and `published`, let's stick with the [example](https://github.com/actions/starter-workflows/blob/8c4fc57de83ba017cd877a7fad3f4dd665c5c4b3/ci/python-publish.yml#L8) for now

**Comment by sfdye** [comment]
Merging this to test the new workflow.

File: .github/workflows/python-publish.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
```
Comment (s-t-e-v-e-n-k) [inline .github/workflows/python-publish.yml:7]: There's a workflow that does most of the heavy lifting: https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/

Comment (sfdye) [reply .github/workflows/python-publish.yml:7]: I am aware, it's too complicated. Let me see if this simple approach works.

**Comment by sfdye** [comment]
It works! on first try 🎉 🎉 🎉 

image

image

I love Github workflow ❤️

**Comment by s-t-e-v-e-n-k** [comment]
Wow! Nice work! Actually, one thing I forgot to point out is we should also update the docs. I'll look at hooking codecov tomorrow and then we can drop travis.

**Comment by sfdye** [comment]
What do you I am doing right now :)

I will leave codecov to you 👍

**Pull request merged**